patternfly/patternfly-elements
# feat: pfe-progress-steps [new component] (#1436)

## <Pull request description>

This PR is meant to serve as a space for reviewing the new pfe-progress-steps item.  Although we are setting it's base as `master` all changes will actually be merged into the `issue-1405-product-trials-epic`.


### Related issues

- (#1436) pfe-progress-steps


### Preview


Link(s) to demo page(s) where this element can be viewed:
- [Link](https://deploy-preview-1613--patternfly-elements.netlify.app/elements/pfe-progress-steps/demo/) 

### Testing instructions

Go to the [demo page](https://deploy-preview-1613--patternfly-elements.netlify.app/elements/pfe-progress-steps/demo/)

- [x] *Dynamically adding stepper items*
  1. Open the console and paste in the following code
  ```
  let item = document.createElement('pfe-progress-steps-item');
  item.innerHTML = `<div slot="title">Bonus</div><a slot="link" href="#">View last step</a>`;
  document.querySelector('pfe-progress-steps').appendChild(item);
  ```
  2. Verify that the first pfe-progress-steps element has 4 steps.
  
- [x] *Test CSS Variables*
  1. Open the console and paste in the follow code:
  ```
  let styles = document.createElement("style");
  styles.innerHTML = `
  :root {
      --pfe-progress-steps-item__circle--color--active: #EE0000;
      --pfe-progress-steps-item__title--Color--active: #151515;
      --pfe-progress-steps__progress-bar--color--active: #6A6E73;
  }`;
  document.head.appendChild(styles);
  ```
  2. Verify that the color of the circle, title color, and progress bar fill color have changed.
  
- [x] *Test vertical state*
  1. Open the console and paste in the follow code:
  ```
    document.querySelector('pfe-progress-steps').setAttribute("vertical", "");
  ```
  2. Verify that the first progress steps are now vertical.

#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist





### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
